### ISSUE TITLE
Adventures in intalling psychopy 1.90.1 on Ubuntu 16.04

### ISSUE THREAD

**Issue by lindeloev** [psychopy/psychopy#1791]
**TL;DR: psychopy module installs and runs just fine on Python 2.7 but the GUI crashes when using `pip install`. After a lot of extra installs and fixes, I can get GUI running using `pip3`. Details below.**

Because of the [outdated neuro.debian package](http://neuro.debian.net/pkgs/psychopy.html), I tried installing psychopy 1.90.1 using `pip`. Installation raised no issues and I can import and use the module just fine in code.

However, when launching the app, `python [path]/psychopyApp.py` shows the splash but then crashes with a segmentation fault after two seconds ([also reported here](https://discourse.psychopy.org/t/good-bad-experience-with-linux-psychopy1-90-0-python3-6/4166)). 

Then I tried installing using `pip3`. Here, wxPython caused problems. First error was `checking for GTK+ - version >= 3.0.0... Package gtk+-3.0 was not found in the pkg-config search path.`. This issue was fixed by installing libgtk-3-dev:

    sudo apt-get install libgtk-3-dev

The `pip3` install then continued but failed with `configure: error: OpenGL libraries not available`. Maybe this issue is specific to older Ubuntu versions (I'm running 16.04) as mentioned [on the wxPython website](https://wxpython.org/pages/downloads/), where they also posted a fix:

    pip install -U -f https://extras.wxpython.org/wxPython4/extras/linux/gtk3/ubuntu-16.04 wxPython

Maybe this would also have fixed the `libgtk-3-dev` issue above? But now running `python3 [path]/psychopyApp.py` fails explicitly:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/psychopy/app/psychopyApp.py", line 96, in <module>
    start_app()
  File "/usr/local/lib/python3.5/dist-packages/psychopy/app/psychopyApp.py", line 28, in start_app
    app = PsychoPyApp(0, showSplash=showSplash)
  File "/usr/local/lib/python3.5/dist-packages/psychopy/app/_psychopyApp.py", line 175, in __init__
    self.onInit(testMode=testMode, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/psychopy/app/_psychopyApp.py", line 205, in onInit
    from psychopy.compatibility import checkCompatibility
  File "/usr/local/lib/python3.5/dist-packages/psychopy/compatibility.py", line 12, in <module>
    import psychopy.data
  File "/usr/local/lib/python3.5/dist-packages/psychopy/data/__init__.py", line 11, in <module>
    from .staircase import (StairHandler, QuestHandler, PsiHandler,
  File "/usr/local/lib/python3.5/dist-packages/psychopy/data/staircase.py", line 28, in <module>
    from psychopy.contrib.psi import PsiObject
  File "/usr/local/lib/python3.5/dist-packages/psychopy/contrib/psi.py", line 27, in <module>
    from scipy import stats
  File "/usr/lib/python3/dist-packages/scipy/stats/__init__.py", line 338, in <module>
    from .stats import *
  File "/usr/lib/python3/dist-packages/scipy/stats/stats.py", line 183, in <module>
    from . import distributions
  File "/usr/lib/python3/dist-packages/scipy/stats/distributions.py", line 10, in <module>
    from ._distn_infrastructure import (entropy, rv_discrete, rv_continuous,
  File "/usr/lib/python3/dist-packages/scipy/stats/_distn_infrastructure.py", line 24, in <module>
    from scipy import optimize
  File "/usr/lib/python3/dist-packages/scipy/optimize/__init__.py", line 233, in <module>
    from ._minimize import *
  File "/usr/lib/python3/dist-packages/scipy/optimize/_minimize.py", line 30, in <module>
    from .lbfgsb import _minimize_lbfgsb
  File "/usr/lib/python3/dist-packages/scipy/optimize/lbfgsb.py", line 44, in <module>
    from scipy.sparse.linalg import LinearOperator
  File "/usr/lib/python3/dist-packages/scipy/sparse/linalg/__init__.py", line 112, in <module>
    from .isolve import *
  File "/usr/lib/python3/dist-packages/scipy/sparse/linalg/isolve/__init__.py", line 6, in <module>
    from .iterative import *
  File "/usr/lib/python3/dist-packages/scipy/sparse/linalg/isolve/iterative.py", line 10, in <module>
    from scipy._lib.decorator import decorator
```

As [adviced here](https://bugs.launchpad.net/ubuntu/+source/python-scipy/+bug/1621803), I then ran

    apt install --reinstall python\*-decorator

... and now psychopy app runs! I ran a Coder demo and a Builder demo successfully. +1 for the python3 support!! But again: an updated neuro.debian would be much more convenient.

**Comment by emailyc** [comment]
is there a 'go-to' guide for installation via Anaconda on Ubuntu 16.04?

**Comment by mirh** [comment]
Is there any reason [sources](https://salsa.debian.org/neurodebian-team/psychopy) are up to 1.90.2, yet [releases](http://neuro.debian.net/pkgs/psychopy.html) aren't moving any further than 1.85.3?

**Comment by peircej** [comment]
The neurodebian (apt-get) installation is no longer up to date or supported - sadly the neurodebian team didn't have time to update it and nor have we yet. To install on linux right now you need to use conda or pip:

https://www.psychopy.org/download.html#linux-install
